agent/consul/state/catalog.go

```diff
@@ -1981,7 +1982,13 @@ func (s *Store) deleteServiceTxn(tx WriteTxn, idx uint64, nodeName, serviceID st
 
 // freeServiceVirtualIP is used to free a virtual IP for a service after the last instance
 // is removed.
-func freeServiceVirtualIP(tx WriteTxn, svc string, excludeGateway *structs.ServiceName, entMeta *acl.EnterpriseMeta) error {
+func freeServiceVirtualIP(
+	tx WriteTxn,
+	svc string,
+	excludeGateway *structs.ServiceName,
+	entMeta *acl.EnterpriseMeta,
+	peerName string,
```

Comment:
Could we pass in structs.PeeredServiceName rather than svc, entMeta, and peerName here?